node/core/backing/src/tests/prospective_parachains.rs

```diff
@@ -1287,14 +1288,19 @@ fn concurrent_dependent_candidates() {
 						break
 					}
 				},
+				AllMessages::StatementDistribution(StatementDistributionMessage::Backed(hash)) => {
+					// Ensure that `Share` was received first for the candidate.
+					assert!(valid_statements.contains(&hash));
+					backed_statements.insert(hash);
+				},
 				_ => panic!("unexpected message received from overseer: {:?}", msg),
 			}
 		}
 
-		assert!(
-			valid_statements.contains(&candidate_a_hash) &&
-				valid_statements.contains(&candidate_b_hash)
-		);
+		assert!(valid_statements.contains(&candidate_a_hash));
+		assert!(valid_statements.contains(&candidate_b_hash));
+		assert!(backed_statements.contains(&candidate_a_hash));
+		assert!(backed_statements.contains(&candidate_b_hash));
```

Comment:
This is the only failing test right now, this line specifically. I'll look into it tomorrow - debugging this will also help solidify my understanding of what's going on.